### ISSUE TITLE
Is there any way of telling if a movie is playable?

### ISSUE THREAD

**Issue by rjx-ray** [bangnoise/ofxHapPlayer#32]
I'm working for Igloovision on updating a Windows app from OF 8.4 to 9.7
We want to use ofxHapPlayer as the default player, but if the movie is non playable we try other players.

I've switched to the libavformat branch and have found that isLoaded() is no longer true immediately after calling load(), I'm guessing because the load happens in a new thread?

Can I find out somehow if the load is in progress and wait for a result?
Is there any other way of testing whether the movie can be played or not?

**Comment by bangnoise** [comment]
Yep, loading happens in the background. This should be clearer, maybe we should add isLoading() or some such.

For the moment, if you have called load() and isLoaded() returns false and getError() returns an empty string, then you can assume it is loading. If you have called load() and getError() returns a non-empty string and isLoaded() returns false, then an error occurred during loading.

**Comment by rjx-ray** [comment]
Thanks for the quick response, I'll try your suggestion.

**Comment by rjx-ray** [comment]
Alternatively, is there any capability to test the codec type before attempting to load?
